crates/symbolicator/src/endpoints/applecrashreport.rs

```diff
@@ -11,6 +15,31 @@ use crate::utils::sentry::ConfigureScope;
 
 use super::ResponseError;
 
+struct MultipartField<'a> {
+    inner: axum::extract::multipart::Field<'a>,
+}
+
+impl<'a> MultipartField<'a> {
+    fn new(inner: axum::extract::multipart::Field<'a>) -> Self {
+        Self { inner }
+    }
+}
+
+impl<'a> Stream for MultipartField<'a> {
+    type Item = Result<bytes::Bytes, std::io::Error>;
```

Comment:
Let me explain why the Newtype around `Field`.  The problem is that `StreamReader` needs a `Stream<Item = Result<T, std::io::Error>` and the native `Field` doesn't do that.  The error itself should be something better obviously, but otherwise I think we can create a new `read_multipart_file` out of this.

I prefer to convert to `AsyncReader` as it allows using `tokio::io::copy` which handles buffering more efficient than building a dumb `if let Some(chunk) = stream.next().await` loop.

For a new `read_multipart_data` function which has filesize limits I think we'll have to use the `if let` loop as I'm not sure what we can do better.